apps/gdalalg_raster_sieve.cpp

```diff
@@ -0,0 +1,245 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster sieve" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_sieve.h"
+
+#include "cpl_conv.h"
+#include "cpl_vsi_virtual.h"
+
+#include "gdal_alg.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "commonutils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*      GDALRasterSieveAlgorithm::GDALRasterSieveAlgorithm()            */
+/************************************************************************/
+
+GDALRasterSieveAlgorithm::GDALRasterSieveAlgorithm()
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL)
+{
+
+    AddProgressArg();
+
+    AddOpenOptionsArg(&m_openOptions);
+    AddInputFormatsArg(&m_inputFormats)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES, {GDAL_DCAP_RASTER});
+    AddInputDatasetArg(&m_inputDataset, GDAL_OF_RASTER);
+
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_RASTER);
+    AddOutputFormatArg(&m_format, /* bStreamAllowed = */ false,
+                       /* bGDALGAllowed = */ false)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES,
+                         {GDAL_DCAP_RASTER, GDAL_DCAP_CREATE});
+    AddCreationOptionsArg(&m_creationOptions);
+    AddOverwriteArg(&m_overwrite);
+
+    auto &mask{AddArg("mask", 0,
+                      _("Use the first band of the specified file as a "
+                        "validity mask (zero is invalid, non-zero is valid)"),
+                      &m_maskDataset)};
+
+    SetAutoCompleteFunctionForFilename(mask, GDAL_OF_RASTER);
+
+    mask.AddValidationAction(
+        [&mask]()
+        {
+            // Check if the mask dataset is a valid raster dataset descriptor
+            // Try to open the dataset as a raster
+            std::unique_ptr<GDALDataset> poDS(
+                GDALDataset::Open(mask.Get<std::string>().c_str(),
+                                  GDAL_OF_RASTER, nullptr, nullptr, nullptr));
+            return static_cast<bool>(poDS);
+        });
+
+    AddBandArg(&m_band);
+    AddArg("size-threshold", 's', _("Minimum size of polygons to keep"),
+           &m_sizeThreshold)
+        .SetDefault(m_sizeThreshold);
+
+    AddArg("connectedness", 'c', _("Consider diagonal pixels as connected"),
```

Comment:
can we change this to be a integer parameter, with values 4 or 8, as in gdalalg_raster_polygonize.cpp ?

Reply:
We can but I guess I'm not the only one who didn't get why a two state toggle takes two integer numbers as arguments. Now I know that 4 and 8 are the adjacent pixels to consider but it took me a while to get there.

Can we find a better naming and turn it into a boolean toggle?

For instance:

--include-diagonal-pixels Include diagonal pixels in the neighbors.

Reply:
@dbaston  Opinion whether --connectedness=4/8 is more or less intuitive than --include-diagonal-pixels ?

The concept of 4-connected/8-connected is at least referenced at https://en.wikipedia.org/wiki/Pixel_connectivity#4-connected / https://en.wikipedia.org/wiki/Pixel_connectivity#8-connected

Reply:
My point is that the argument is not an integer but a boolean because it only has two possible states. A native English speaker will probably find the best way to put the concept of diagonal pixel inclusion into understandable words.

Reply:
An image would make it understandable easily - are these connected or not?
<img width="103" alt="image" src="https://github.com/user-attachments/assets/650d568a-9935-496d-98c4-2ae5d2e5c475" />
I do not know how to make it understandable for beginning users with few words. I fear that `--connectedness=edge/corner` is not evident for everybody either. The best option so far is  --include-diagonal-pixels for my mind.

Reply:
I agree that a flag is more clear. `--include-diagonal-pixels` or maybe `--connect-diagonal-pixels`

Reply:
`--connect-diagonal-pixels` is good and sounds with "connectedness".

Reply:
I have changed to `--connect-diagonal-pixels` and expanded the documentation, I hope it is clearer now.